docs/deploy-rbd.md

```diff
@@ -63,6 +63,7 @@ make image-cephcsi
 | `csi.storage.k8s.io/provisioner-secret-namespace`, `csi.storage.k8s.io/node-stage-secret-namespace` | yes (for Kubernetes) | namespaces of the above Secret objects                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
 | `mounter`                                                                                           | no                   | if set to `rbd-nbd`, use `rbd-nbd` on nodes that have `rbd-nbd` and `nbd` kernel modules to map rbd images                                                                                                                                                                                                                                                                                                                                                                                             |
 | `encrypted`                                                                                         | no                   | disabled by default, use `"true"` to enable LUKS encryption on PVC and `"false"` to disable it. **Do not change for existing storageclasses**                                                                                                                                                                                                                                                                                                                                                          |
+| `encryptionType`                                                                                    | no                   | encryption type to use, currently supported types are luks1 and luks2 (defaults to luks2)                                                                                                                                                                                                                                                                                                                                                                                                               |
```

Comment:
It's the other way around. luks1 format remains fully supported. There should be a way to choose b/w available formats, hence it doesn't hurt to be flexible (when cryptosetup and kernel are ok with it, why not cephcsi?). Who knows in the future there could be other types (luks3 or whatever) :-)

Reply:
Adding more features adds to the maintenance burden, so I prefer to only add features when needed.

What version of LUKS is currently the default or standard? If LUKS2 is still new and LUKS1 is more common to be used, then we would need to support both. However, if LUKS1 is mostly replaced by LUKS2, then I do not see a need to support LUKS1.

Reply:
@nixpanic Cryptsetup currentrly supports different encryption operating modes to use with dm-crypt:

```
--type luks for using the default LUKS format version (LUKS1 with cryptsetup < 2.1.0, LUKS2 with cryptsetup ≥ 2.1.0),
--type luks1 for using LUKS1, the most common version of LUKS,
--type luks2 for using LUKS2, the latest available version of LUKS that allows additional extensions,
--type plain for using dm-crypt plain mode,
--type loopaes for a loopaes legacy mode,
--type tcrypt for a TrueCrypt compatibility mode.
```

We only tested ceph-csi with luks1 and luks2, so far. Also, its worth noting that luks1 remains stable, supported "forever"
Having these changes makes us flexible to opt b/w various format types.

Reply:
I am mostly interested by the default version of LUKS used by cryptsetup on the distributions that we support with Ceph-CSI. We build on CentOS-8 now, and I guess if that provides LUKS2 by default, we do not need to add LUKS1.

The additional tests, and maintenance to make sure we don't break LUKS1 support might not be worth the benefit. If LUKS1 is the current default and we expect users to have provisioned volumes with it, we probably need to maintain support for it, unfortunately.

If there ever will be a need to support different encryption operation modes, I don't know, but I hope single one will be sufficient for us.

Reply:
Well, I really didn't follow why we shouldn't support other encryption formats supported by cryptsetup and dm_encrypt, after all Ceph-CSI is operating at the very top of the stack.

Yes, this can be a feature request If that makes us happy here :-)

Will remove this commit in the later respin.